### DISCUSSION TITLE
How to inject data like classes on/before ssr on document root or body

### DISCUSSION THREAD

**Discussion by sharmapukar217** [rakkasjs/rakkasjs#73] [Q&A] [answered]
To inject something on the head there's emitToDocumentHead hook but I need to inject classname to body or document on ssr. Is there any method to do so?

**Comment by cyco130** [comment]
You can use the `emitBeforeSsrChunk` with a boolean flag to only inject at the very beginning of the stream: https://stackblitz.com/edit/github-n4gfag?file=src%2Froutes%2Findex.page.jsx,src%2Fentry-hattip.js

**Reply by sharmapukar217** [reply]
But this won't work if I want to add classes to body

**Reply by sharmapukar217** [reply]
Oh I got it now. Returning <html class="some class"> from emitBeforeSsrChunks worked 😊

**Comment by cyco130** [comment, marked as answer]
Oh, sorry, I got it all wrong! I think you may make it work but it's probably invalid HTML.

You're right, [there isn't a way yet](https://github.com/rakkasjs/rakkasjs/blob/2006e0362db20269d5c61d8473429350a3ee36f7/packages/rakkasjs/src/features/pages/middleware.tsx#L504).

It will work with either an HTML shell customization hook or with the `Head` component (and returning `head` from `prerender`) but it's not implemented yet.

**Reply by sharmapukar217** [reply]
> Oh, sorry, I got it all wrong! I think you may make it work but it's probably invalid HTML.
Is it an invalid html? But even on inspecting the page source, everything seems good. If it's invalid html then I'll see if any error occurs otherwise its working for me ( just used to set theme saved on cookie )

**Reply by sharmapukar217** [reply]
Yes you're right. Its totally invalid. Doing so is injecting new html document inside body 😅
